docs/dev/optimizer.md

```diff
@@ -0,0 +1,55 @@
+
+The security of the calculation is guaranteed to a required level, typically 128 bits. The correctness of the calculation is guaranteed up to a given error rate. The efficiency is optimized to reduce the time to execute the FHE calculus.
+
+These quantities, known as crypto-parameters, only exist due to the FHE nature of the calculus. The crypto-parameters are in a finite but huge space. The role of the optimizer is to quickly find the most efficient crypto-parameters possible while guaranteeing security and correctness.
+
+## Security, Correctness, and Efficiency
+
+### Security
+
+The security level is chosen by the user.
+
+We typically operate at a fixed security level, such as 128 bits, to ensure that there is never a trade-off between security and efficiency. This constraint imposes a minimum amount of noise in any calculation and in public material.
+
+An independent public research tool, the lattice estimator, estimates the security level.
+
+The lattice estimator is a joint work from FHE experts. It estimates the security level of the couple noise and the mask size of the ciphertext (i.e. one of the crypto-parameters).
+
+For each security level, a parameter curve of the appropriate minimal error level is pre-computed using the lattice estimator and used by the optimizer. Learn more about the parameter curve [here](https://www.notion.so/tools/parameter-curves/README.rst).
+
+### Correctness
+
+Correctness decreases as the level of noise increases.
+
+Noise accumulates during calculations until it is actively reduced. Too much noise can render the calculation's result inaccurate or, worse, completely incorrect.
+
+Before optimization, we compute a noise bound that guarantees a given error level if the noise always stays under that bound before we reduce it or before the result is decrypted. The noise growth depends on a critical quantity: the norm2 of any dot product (or equivalent) present in the calculus. This norm2 changes the scale of the noise, so we must reduce it sufficiently for the next dot product operation whenever we reduce the noise.
+
+The user can choose the degree of correctness using two options: the PBS error probability and the global error probability.
+
+The PBS error probability controls correctness locally, while the global error probability focuses on the overall calculation result. In the end, they both control the same elements, but one option may be easier to use depending on the specific use case.
+
+### Efficiency
+
+Efficiency decreases as more noise must be reduced.
+Efficiency decreases as more precision are required, e.g. 7bits versus 8bits.
+
+The cost of noise reduction increases as the scale of the noise reduction increases. The larger the norm2, the more the noise will grow after noise reduction, and the greater the cost will be to reduce it sufficiently to stay within the noise bound.
```

Comment:
> The cost of noise reduction increases as the scale of the noise reduction increases. The larger the norm2, the more the noise will grow after noise reduction, and the greater the cost will be to reduce it sufficiently to stay within the noise bound.

The larger the 2norm is, the bigger noise will be after a dot product. To still remain below the noise bound, the inputs of the dot product must have a noise sufficiently small. The smaller this noise is, the slower the previous bootstrapping are. Therefore, the larger the 2norm is, the slower the computation will be.